watcher/src/watcher_db.rs

```diff
@@ -163,14 +163,72 @@ impl WatcherDB {
         }
     }
 
-    /// Get the total number of Blocks in the watcher db.
+    /// Get the last synced block per url for a given set of urls
     pub fn last_synced_blocks(
         &self,
         src_urls: &[Url],
     ) -> Result<HashMap<Url, Option<u64>>, WatcherDBError> {
         let db_txn = self.env.begin_ro_txn()?;
-        let mut results = HashMap::default();
+        self.get_url_to_last_synced(src_urls, &db_txn)
+    }
+
+    /// In the case where a synced block did not have a signature, update last synced.
+    pub fn update_last_synced(
+        &self,
+        src_url: &Url,
+        block_index: u64,
+    ) -> Result<(), WatcherDBError> {
+        let mut db_txn = self.env.begin_rw_txn()?;
+        db_txn.put(
+            self.last_synced,
+            &src_url.as_str().as_bytes(),
+            &block_index.to_be_bytes(),
+            WriteFlags::empty(),
+        )?;
+        db_txn.commit()?;
+        Ok(())
+    }
+
+    /// Get the highest block that all urls have synced.
+    /// Note: In the case where one watched consensus validator dies and is no longer
```

Comment:
I'd love to see this generalized someday to allow a quorum set to define "synced"

Reply:
That's a good suggestion. The other thing I'd like to see on the roadmap for the watcher is to have some sense of "discovery" of the quorum, so that it could always be guaranteed to get a Signature + Timestamp from the nodes that participated in consensus. Currently, it's possible that the watcher is configured to watch a set of nodes that ends up offline, and the watcher would have missing timestamps during those blocks.

Reply:
For now, "synced" in this context means per-URL, how many blocks has it synced.

Reply:
Restarting the watcher won't help since old urls are not removed.

Reply:
Yeah, it won't help for this new call, but it will help for the old call used by WatcherSyncThread, which would then only get highest synced for the newly watched URLs and ignore the old one.